File: app/src/main/java/ru/samtakoy/core/screens/log/MyLog.java
package ru.samtakoy.core.screens.log;

import android.util.Log;

import java.util.Collections;
import java.util.LinkedList;
import java.util.List;

public class MyLog {


    private static List<String> sLog = new LinkedList<>();

    public static void add(String s){
        sLog.add(s);
        Log.e("MyLog", s);
    }

    public static List<String> getStrings(){
        return Collections.unmodifiableList(sLog);
    }
}

File: app/src/main/java/ru/samtakoy/core/presentation/log/MyLog.kt
package ru.samtakoy.core.presentation.log

import android.util.Log
import org.apache.commons.lang3.exception.ExceptionUtils
import java.util.Collections
import java.util.LinkedList

object MyLog {
    private val sLog: MutableList<String> = LinkedList<String>()

    @JvmStatic
    fun add(source: String, t: Throwable?) {
        val s = source + ": " + t?.message + "\n" + t?.stackTraceToString()
        sLog.add(s)
        Log.e("MyLog", s)
    }

    @JvmStatic
    fun add(s: String) {
        sLog.add(s)
        Log.e("MyLog", s)
    }

    @JvmStatic
    val strings: MutableList<String>
        get() = Collections.unmodifiableList<String>(sLog)
}
